src/mpid/ch4/netmod/ofi/init_settings.c

```diff
@@ -0,0 +1,357 @@
+     * index */
+    if (MPIDI_OFI_ENABLE_SCALABLE_ENDPOINTS)
+        hints->caps |= FI_NAMED_RX_CTX;
+
+    /* ------------------------------------------------------------------------ */
+    /* Set object options to be filtered by getinfo                             */
+    /* domain_attr:  domain attribute requirements                              */
+    /* op_flags:     persistent flag settings for an endpoint                   */
+    /* endpoint type:  see FI_EP_RDM                                            */
+    /* Filters applied (for this netmod, we need providers that can support):   */
+    /* THREAD_DOMAIN:  Progress serialization is handled by netmod (locking)    */
+    /* PROGRESS_AUTO:  request providers that make progress without requiring   */
+    /*                 the ADI to dedicate a thread to advance the state        */
+    /* FI_DELIVERY_COMPLETE:  RMA operations are visible in remote memory       */
+    /* FI_COMPLETION:  Selective completions of RMA ops                         */
+    /* FI_EP_RDM:  Reliable datagram                                            */
+    /* ------------------------------------------------------------------------ */
+    hints->addr_format = FI_FORMAT_UNSPEC;
+    hints->domain_attr->threading = FI_THREAD_DOMAIN;
+    MPIDI_OFI_set_auto_progress(hints);
+    hints->domain_attr->resource_mgmt = FI_RM_ENABLED;
+    hints->domain_attr->av_type = MPIDI_OFI_ENABLE_AV_TABLE ? FI_AV_TABLE : FI_AV_MAP;
+
+    if (fi_version() >= FI_VERSION(1, 5)) {
+        hints->domain_attr->mr_mode = 0;
+#ifdef FI_RESTRICTED_COMP
+        hints->domain_attr->mode = FI_RESTRICTED_COMP;
+#endif
+
+#ifdef FI_MR_VIRT_ADDR
+        if (MPIDI_OFI_ENABLE_MR_VIRT_ADDRESS) {
+            hints->domain_attr->mr_mode |= FI_MR_VIRT_ADDR;
+        }
+#endif
+
+#ifdef FI_MR_ALLOCATED
+        if (MPIDI_OFI_ENABLE_MR_ALLOCATED) {
+            hints->domain_attr->mr_mode |= FI_MR_ALLOCATED;
+        }
+#endif
+
+#ifdef FI_MR_PROV_KEY
+        if (MPIDI_OFI_ENABLE_MR_PROV_KEY) {
+            hints->domain_attr->mr_mode |= FI_MR_PROV_KEY;
+        }
+#endif
+    } else {
+        /* In old versions FI_MR_BASIC is equivallent to set
+         * FI_MR_VIRT_ADDR, FI_MR_PROV_KEY, and FI_MR_ALLOCATED on.
+         * FI_MR_SCALABLE is equivallent to all bits off in newer versions.
+         */
+        MPIR_Assert(MPIDI_OFI_ENABLE_MR_VIRT_ADDRESS == MPIDI_OFI_ENABLE_MR_PROV_KEY);
+        if (MPIDI_OFI_ENABLE_MR_VIRT_ADDRESS) {
+            hints->domain_attr->mr_mode = FI_MR_BASIC;
+        } else {
+            hints->domain_attr->mr_mode = FI_MR_SCALABLE;
+        }
+    }
+    /* FI_MR_SCALABLE is implied by lack of mr_mode bits in >= v1.5.
+     * But at least sockets provider still need FI_MR_SCALABLE to be set
+     * or it will return FI_MR_BASIC. */
+    if (hints->domain_attr->mr_mode == 0) {
+        hints->domain_attr->mr_mode = FI_MR_SCALABLE;
```

Comment:
@wesbland If I delete this line, `fi_getinfo` using this hint (with `mr_mode=0`) will return provider with `mr_mode=FI_MR_BASIC`, i.e. opposite to what we want unless we set `mr_mode` to `FI_MR_SCALABLE`. This is with current embedded libfabric.

Reply:
Strange. I guess I'm not going to die on this hill...